### PR TITLE
Implement text props name convention

### LINKED ISSUE #1429: Add built-in localization support
We should support new localization feature. 

- [ ] Rename props to look conventional to material ui text props :
If we should rename the prop to `titleText` to something that starts to look like a convention:
- backIconButtonText
- labelRowsPerPage
- labelDisplayedRows
- nextIconButtonText
- getLabelText
- clearText
- closeText
- loadingText
- noOptionsText
- openText

_Originally posted by @oliviertassinari in https://github.com/_render_node/MDE3OlB1bGxSZXF1ZXN0UmV2aWV3MzM1MDA1Njky/pull_request_reviews/more_threads_

- [ ] Make it possible to localize all formats from the localization feature. It should be a killer feature

### PR DESCRIPTION
**Pull request by dmtrKovalenko**
This PR resolves the first part of #1654 and probably closes #1429   

One question as for me. For now, all of our text props are spread through the different components, like `PickersModalDialog`, `ArrowSwitcher`, and `KeyboardDateInput`. It looks like that will be too awful to localize. 

Maybe we should add one prop like `texts` with all the text props combined in 1 object? But it is still possible to pass any prop to the just `DatePicker`, so support localization only for root components could be a workaround as well

### CODE REVIEW

**Comment by oliviertassinari** [comment]
Regarding #1429, I would propose we close it once we have the counterpart of the localization strings in the mono repository, probably with a working demo in https://material-ui.com/guides/localization/#example. I think that we need to make sure it works, end-to-end. It would also be a great constrain to avoid regressions. What do you think?

> Maybe we should add one prop like texts with all the text props combined in 1 object?

Agree, I think that after 5 localization strings for a single component, we should start to wonder. With 10, a `texts` likrbprop sounds required. Should we call it `texts` or something else? 

There is one aspect I wonder about. How should we handle the strings duplication and fragmentation? Do you have an example of what the change to `locale.ruRu` look like to have to have the date picker components localized in Russian?

**Comment by dmtrKovalenko** [comment]
I propose the following strategy:
1. Merge this PR with only renaming prop
2. Work for another PR with default props support (#1340)
3. Work on Localization PR with examples and support (and there combine props into new object)

**Comment by oliviertassinari** [comment]
@dmtrKovalenko This plan sounds awesome!

File: lib/src/__tests__/DatePickerProps.test.tsx [hunk 1]
```diff
@@ -63,15 +63,15 @@ describe('DatePicker - different props', () => {
...
+  it('showTodayButton – accept current date when "today" button is clicked', () => {
     const onCloseMock = jest.fn();
     const onChangeMock = jest.fn();
     const component = mount(
       <MobileDatePicker
         renderInput={props => <TextField {...props} />}
         autoOk
         showTodayButton
-        cancelLabel="stream"
+        cancelText="stream"
```
Comment (oliviertassinari) [inline lib/src/__tests__/DatePickerProps.test.tsx:74]: Any reason for providing a string in this test?

File: lib/src/_shared/PickerModalDialog.tsx [hunk 2]
```diff
@@ -7,21 +7,48 @@ import Dialog, { DialogProps } from '@material-ui/core/Dialog';
...
-  onDismiss: () => void;
-  onClear: () => void;
-  onSetToday: () => void;
-  okLabel?: React.ReactNode;
-  cancelLabel?: React.ReactNode;
-  clearLabel?: React.ReactNode;
-  todayLabel?: React.ReactNode;
+export interface ExportedPickerModalProps {
+  /**
+   * "OK" button text
```
Comment (oliviertassinari) [inline lib/src/_shared/PickerModalDialog.tsx]: Maybe should make a sentence.

```suggestion
   * "OK" button text.
```

File: lib/src/_shared/PickerModalDialog.tsx [hunk 3]
```diff
@@ -7,21 +7,48 @@ import Dialog, { DialogProps } from '@material-ui/core/Dialog';
...
+   * @default "TODAY"
+   */
+  todayText?: React.ReactNode;
+  /**
+   * Show clear action in picker dialog
+   * @default false
+   */
   clearable?: boolean;
+  /**
+   * If `true` today button will be displayed. **Note** that `showClearButton` has higher priority
```
Comment (oliviertassinari) [inline lib/src/_shared/PickerModalDialog.tsx]:
```suggestion
   * If `true`, the today button will be displayed. **Note** that `showClearButton` has a higher priority.
```

File: lib/src/_shared/PickerModalDialog.tsx [hunk 4]
```diff
@@ -7,21 +7,48 @@ import Dialog, { DialogProps } from '@material-ui/core/Dialog';
...
+   * @default "CLEAR"
+   */
+  clearText?: React.ReactNode;
+  /**
+   * "TODAY" Text message
+   * @default "TODAY"
+   */
+  todayText?: React.ReactNode;
+  /**
+   * Show clear action in picker dialog
```
Comment (oliviertassinari) [inline lib/src/_shared/PickerModalDialog.tsx]:
```suggestion
   * If `true`, it shows the clear action in the picker dialog.
```

File: lib/src/__tests__/DatePickerProps.test.tsx (hunk 1 above)
Comment (dmtrKovalenko) [reply lib/src/__tests__/DatePickerProps.test.tsx:74]: It is probably not the best way, but I am testing typescript bindings for props such way. 
Probably need to add some special kind of typescript definition tests

Comment (oliviertassinari) [reply lib/src/__tests__/DatePickerProps.test.tsx:74]: So the purpose is to have a test equivalent to, say, https://github.com/mui-org/material-ui/blob/a75c19f267edfdf71487764961de5c425096f47b/packages/material-ui/src/Typography/typography.spec.tsx?

**Comment by oliviertassinari** [comment]
@dmtrKovalenko We have an ongoing thread on the naming convention for these props in https://github.com/mui-org/material-ui/issues/21018#issuecomment-629701596.

**Pull request merged**